src/Examples/Flowsheets/power_generation/rsofc/cpu.py

```diff
@@ -0,0 +1,720 @@
+                b.inlet_mole_frac_comp[t, "H2O"],
+                b.inlet_mole_frac_comp[t, "N2"],
+            )
+
+        # Vent stream: Pressure
+        @self.Constraint(self.flowsheet().config.time, doc="Pressure: vent [Pa]")
+        def vent_pressure_eq(b, t):
+            return b.vent_pressure[t] == sm.vent_pressure_fun.f(
+                b.inlet_flow_mol[t],
+                b.inlet_mole_frac_comp[t, "Ar"],
+                b.inlet_mole_frac_comp[t, "CO2"],
+                b.inlet_mole_frac_comp[t, "O2"],
+                b.inlet_mole_frac_comp[t, "H2O"],
+                b.inlet_mole_frac_comp[t, "N2"],
+            )
+
+    def initialize(blk, outlvl=idaeslog.NOTSET, solver="ipopt", optarg={"tol": 1e-6}):
+        """
+        CO2 pure pyomo block initialization routine
+
+        Keyword Arguments:
+            outlvl : sets output level of initialisation routine
+
+            optarg : solver options dictionary object (default={'tol': 1e-6})
+            solver : str indicating whcih solver to use during
+                     initialization (default = 'ipopt')
+
+        Returns:
+            None
+        """
+        iscale.calculate_scaling_factors(blk)
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+        opt = pyo.SolverFactory(solver)
+        opt.options = optarg
+
+        init_log.info_low("Starting initialization...")
+
+        blk.inlet.flow_mol[0].fix()
+        blk.inlet.mole_frac_comp[0, "Ar"].fix()
+        blk.inlet.mole_frac_comp[0, "CO2"].fix()
+        blk.inlet.mole_frac_comp[0, "O2"].fix()
+        blk.inlet.mole_frac_comp[0, "H2O"].fix()
+        blk.inlet.mole_frac_comp[0, "N2"].fix()
+
+        # solve model
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+        init_log.info_high("Initialization Step 1 {}.".format(idaeslog.condition(res)))
+        init_log.info_high("Initialization Step 1 Complete.")
+
+        # check component material balances
+        # deactivate component balance equation for inlet
+        # fix all inlet flows
+        # blk.inlet.mole_frac_comp[0, 'O2'].fix(0)
+        # blk.mole_frac_comp_inlet_eqn.deactivate()
+        # with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+        #     res = opt.solve(blk, tee=slc.tee)
+        # init_log.info_high(
+        #         "Initialization Step 2 {}.".format(idaeslog.condition(res))
+        #     )
+
+        # release all states and activate constraint
+        blk.inlet.flow_mol[0].unfix()
+        blk.inlet.mole_frac_comp[0, "Ar"].unfix()
+        blk.inlet.mole_frac_comp[0, "CO2"].unfix()
+        blk.inlet.mole_frac_comp[0, "O2"].unfix()
+        blk.inlet.mole_frac_comp[0, "H2O"].unfix()
+        blk.inlet.mole_frac_comp[0, "N2"].unfix()
+        # blk.mole_frac_comp_inlet_eqn.activate()
+
+        init_log.info("Initialization Complete.")
```

Comment:
There should be a check for optimal termination here which raises an Initialization error if it fails.